distributions/test/src/test/java/com/connexta/ion/replication/distributions/test/ReplicationITest.java

```diff
@@ -0,0 +1,144 @@
+package com.connexta.ion.replication.distributions.test;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableMap;
+import com.jayway.restassured.path.xml.XmlPath;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+// TODO: why does this need to end in "test" for the tests to run?
+public class ReplicationITest extends IntegrationTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ReplicationITest.class);
+
+  private static final Slf4jLogConsumer logConsumer = new Slf4jLogConsumer(LOGGER);
+
+  private Set<String> idsCreated = new HashSet<>();
+
+  @BeforeClass
+  public static void beforeClass() {
```

Comment:
If a container has an exposed port, Testcontainers will wait for that port to be responive by default. We can also create our own readyness checks. ATM we just use the default readyness check to make sure solr is up. I don't have a readyness check for replication because it should be the last thing we're waiting on. If it doesn't come up in time for one of the tests to pass I don't think it's going to come up at all. If we want to add a readyness check later, we can use the spring-actuator included in #144 to implement a health check for us.

Reply:
Being responsive doesn't mean the same as ready.

Reply:
I added a wait strategy to solr that waits for the config core to be ready and I added one to replication that waits for the "started main in ____ seconds" log before proceeding with the tests

Reply:
Good. Although we should avoid relying on logs. Could we do or enhance the system such that we would be able to better make that determination as opposed to looking at logs?